### PR TITLE
feat(scripts/bittensor): add bittensor recovery tool

### PR DESCRIPTION
**Pull request by JacobPlaster**
#### Summary
This PR introduces a new script under `scripts/bittensor-recovery-script` to facilitate key recovery and fund transfers on the Bittensor network. The script enables users to derive Ed25519 keys from raw private scalars and execute transactions seamlessly.

```
$ cd scripts/bittensor-recovery-script
$ npm install && npm start
```

#### Key Features
- **Ed25519 Key Recovery**:  
  Derives public keys and constructs 64-byte secret keys from raw private scalars without clamping, ensuring compatibility with MPC outputs.
- **Transaction Creation and Broadcasting**:  
  Creates, signs, and submits `transferKeepAlive` transactions to the Bittensor network.
- **Input Validation**:  
  - Private Key: Ensures a valid 32-byte Ed25519 scalar (64 hex characters).  
  - Destination Address: Validates SS58 address format.  
  - Transfer Amount: Ensures positive numeric inputs.
- **User Prompts**:  
  Guides users through the recovery and transaction process interactively.
- **Default Endpoint**:  
  Supports `wss://entrypoint-finney.opentensor.ai:443` for network connectivity.

#### Usage
1. Enter the Ed25519 private key scalar (64 hex chars).  
2. Provide the SS58 destination address and transfer amount.  
3. Optionally configure the Bittensor endpoint.  
4. The script will derive the keypair, sign, and submit the transaction.

### CODE REVIEW

File: scripts/bittensor-recovery-script/main.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  const api = await ApiPromise.create({ provider });
+
+  console.log("Creating keyring...");
+  const keyring = new Keyring({ type: 'ed25519', ss58Format: SS58_FORMAT });
+
+  // Prepare the private key
+  let privateKeyBytes = Buffer.from(privateKey, 'hex');
+
+  // Derive the public key from the private key
+  const derivedPublicKeyBytes = await ed.getPublicKey(privateKeyBytes);
```
Comment (JacobPlaster) [inline scripts/bittensor-recovery-script/main.ts]: @notatestuser this code doesnt seem to derive the same public key as I'd expect it to. James's vault (`m9v62s7b7114nqs7tf421mx0`) has the address `5DDRNSii3jUdb4yWot6ZvzeXg3DtUG7vWXQzVJkbVnZKYqUG` and when I run the script using his EdDSA/Ed25519 private key from the DR tool ([slack canvas for DR tool export](https://iofinnet.slack.com/docs/T02PB4D04P5/F085AL6PCSJ)) im getting a different address. Signature seems to be accepted ok though....

My question is are there any obvious mess ups here? Perhaps im missing out some manipulations that TSS performs?

Comment (JacobPlaster) [reply scripts/bittensor-recovery-script/main.ts]: I've been down the rabbit hole of deriving the keys without 'clamping', and also ensuring the correct padding on the privateKey buffer. They did derive different public addresses but not the address I was expecting

Comment (notatestuser) [reply scripts/bittensor-recovery-script/main.ts]: Two ways to do it.

1. Ask them to input the public key from the DR tool https://github.com/IoFinnet/io-vault-disaster-recovery-cli/blob/master/scripts/xrpl-tool/main.ts#L37
2. Derive it yourself in the script using this way https://github.com/IoFinnet/io-vault-disaster-recovery-cli/blob/master/scripts/xrpl-tool/main.ts#L207

Comment (notatestuser) [reply scripts/bittensor-recovery-script/main.ts]: Want me to give that a try @JacobPlaster or are you on it?

Comment (JacobPlaster) [reply scripts/bittensor-recovery-script/main.ts]: Well I did try accepting the public key as an input, and that did indeed derive the correct expected address. But then i'd get invalid signature.

I'd be very greatful if you could take a whack at it, I've sunk quite a bit of time into it going down the chatGPT rabbit hole to remove clamping, adding padding, different crypto libraries ect...

Comment (notatestuser) [reply scripts/bittensor-recovery-script/main.ts]: Do you have that version where you loaded the key by input? I would imagine it was just using the address in a wrong way

Comment (notatestuser) [reply scripts/bittensor-recovery-script/main.ts]: If not, I can work with this

Comment (notatestuser) [reply scripts/bittensor-recovery-script/main.ts]: Try that with the latest commit @JacobPlaster

**Review by notatestuser — DISMISSED** [review]

**Review by notatestuser — DISMISSED** [review]

**Review by notatestuser — APPROVED** [review]

**Pull request merged**